assets/training/model_management/src/azureml/model/mgmt/processors/transformers/convertors.py

```diff
@@ -193,29 +196,58 @@ def _save(
             logger.info("Experimental features enabled for MLflow conversion")
             self._hf_conf["exp"] = True
 
-        hf_mlflow.hftransformers.save_model(
-            config=config,
-            tokenizer=tokenizer,
-            hf_model=model,
-            hf_conf=self._hf_conf,
-            conda_env=conda_env,
-            code_paths=code_paths,
-            signature=self._signatures,
-            input_example=input_example,
-            requirements_file=requirements_file,
-            pip_requirements=pip_requirements,
-            extra_pip_requirements=self._extra_pip_requirements,
-            path=self._output_dir,
-        )
+        try:
+            # create a conda environment for OSS transformers Flavor
+            python_version = platform.python_version()
+            pip_pkgs = self._get_curated_environment_pip_package_list()
+            conda_deps = CondaDependencies.create(conda_packages=None,
+                                                  python_version=python_version,
+                                                  pip_packages=pip_pkgs,
+                                                  pin_sdk_version=False)
+            curated_conda_env = conda_deps.as_dict()
+
+            model_pipeline = transformers.pipeline(task=self._task, model=model)
+
+            mlflow.transformers.save_model(
```

Comment:
what if this fails with OOM? Are we expecting customers to provide compute sufficient enough to load the model? Should we fail before in this scenario?

Reply:
Maybe we can handle this in the validation component?

Reply:
loading with device_map="auto"

Reply:
using device_map="auto" can also lead to OOM in cases when combined memory is short for a model? Please let me know if this not correct. 

Don't think this is a blocker per se, but we need to handle this and make sure we throw a safe exception here

Reply:
we can device_map="auto". I was saying that even with that, there is a chance that model would not be loaded. How are we dealing with that scenario? We should fail it safely by estimating model memory req and what is available